File: 70.climbing-stairs.cs
// @leet start
public class Solution {
    public int ClimbStairs(int n) {
      if(n == 1) return 1;
      var a = 1;
      var b = 2;
      for(int i = 3; i <= n; i++) {
        b = a + b;
        a = b - a;
      }
      return b;
    }
}
// @leet end

File: 70.climbing-stairs.kt
// @leet start
class Solution {
    fun climbStairs(n: Int): Int {
      if (n <= 2) return n
      var a = 1
      var b = 2
      for (i in 3..n) {
        b = a + b
        a = b - a
      }
      return b
    }
}
// @leet end
